server/protocol/qdatastream.py

```diff
@@ -95,7 +95,14 @@ async def read_message(self):
         if action in ("PING", "PONG"):
             return {"command": action.lower()}
 
-        message = json.loads(action)
+        try:
+            message = json.loads(action)
+        except json.decoder.JSONDecodeError as e:
+            raise json.decoder.JSONDecodeError(
+                msg = f"Invalid JSON, full action string was: \
+                    {action[e.pos-100:e.pos+100]}",
+                doc = e.doc,
+                pos = e.pos) from e
```

Comment:
I'm surprised codacy isn't complaining about this but flake8 doesn't like the spaces around `=` here. Can we format it like this?
```suggestion
                msg=f"Invalid JSON, near: {action[e.pos-100:e.pos+100]}",
                doc=e.doc,
                pos=e.pos
            ) from e
```

Reply:
Will do, but the unit test will take some time, having some trouble with the dev environment.

Reply:
Are you in Windows? If you message me on the faf discord I can try to help you out.

You can also just make a change, push, and look at the github actions output. Then rebase once everything works.